src/components/Accordion/index.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useRef, useState } from "react";
+import styled from "@emotion/styled";
+import Icon from "@leafygreen-ui/icon";
+import { size } from "constants/tokens";
+
+interface AccordionProps {
+  "data-cy"?: string;
+  defaultOpen?: boolean;
+  children: React.ReactNode;
+  className?: string;
+  onToggle?: (s: { isVisible: boolean }) => void;
+  subtitle?: React.ReactNode;
+  title: React.ReactNode;
+  titleTag?: React.FC;
+  toggledTitle?: React.ReactNode;
+}
+
+const Accordion: React.FC<AccordionProps> = ({
+  "data-cy": dataCy,
+  defaultOpen = false,
+  children,
+  className,
+  onToggle = () => {},
+  subtitle,
+  title,
+  titleTag,
+  toggledTitle,
+}) => {
+  const [isAccordionDisplayed, setIsAccordionDisplayed] = useState(defaultOpen);
+
+  const showToggledTitle = isAccordionDisplayed ? toggledTitle : title;
+  const TitleTag = titleTag ?? "span";
+  const titleComp = (
+    <TitleTag>{toggledTitle ? showToggledTitle : title}</TitleTag>
+  );
+
+  const childrenRef = useRef<HTMLDivElement>(null);
+  const [childrenHeight, setChildrenHeight] = useState(0);
+
+  useEffect(() => {
+    if (childrenRef && childrenRef.current) {
+      setChildrenHeight(childrenRef.current.scrollHeight);
+    }
+  }, [children, childrenRef]);
+
+  const toggleAccordionHandler = () => {
+    setIsAccordionDisplayed(!isAccordionDisplayed);
+    onToggle({ isVisible: !isAccordionDisplayed });
+  };
+
+  return (
+    <div className={className} data-cy={dataCy}>
```

Comment:
What do you think about using `<details>` and `<summary>` to introduce some more semantic HTML? ((docs)[https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details])
```suggestion
    <details className={className} data-cy={dataCy}>
```

Reply:
I wanted this to work so badly  but then I ran into a problem where I couldn't get it to animate nicely 😔 The issue is also described in the docs:

<img width="565" alt="Screenshot 2023-06-16 at 2 52 22 PM" src="https://github.com/evergreen-ci/parsley/assets/47064971/c23a660d-189b-42b0-a58f-99ddecbb5c00">

I think it's not too bad to keep it as a div since we have `role="button"` and `aria-expanded` attributes, but let me know if any other accessibility attributes come to mind that I should add!

Reply:
That's a shame, also cool find @sophstad! didn't know this existed.